newroutes/collectionPages.js

```diff
@@ -1,134 +1,158 @@
 const express = require("express")
 
 // Import middleware
+const { BadRequestError } = require("@errors/BadRequestError")
+
 const {
   attachReadRouteHandlerWrapper,
   attachWriteRouteHandlerWrapper,
   attachRollbackRouteHandlerWrapper,
 } = require("@middleware/routeHandler")
 
-// Import classes
-const CollectionController = require("@controllers/CollectionController")
-
 const {
   CreatePageRequestSchema,
   UpdatePageRequestSchema,
   DeletePageRequestSchema,
-} = require("./RequestSchema")
-
-const router = express.Router()
-
-// Create new page in collection
-async function createCollectionPage(req, res) {
-  const { accessToken } = req
-
-  const { siteName, collectionName, subcollectionName } = req.params
-  const { error } = CreatePageRequestSchema.validate(req.body)
-  if (error) throw error
-  const { newFileName, pageBody, frontMatter } = req.body
-  const createResp = await CollectionController.CreatePage(
-    { siteName, accessToken },
-    {
-      fileName: newFileName,
-      collectionName,
-      content: pageBody,
-      frontMatter,
-      thirdNavTitle: subcollectionName,
-    }
-  )
-
-  return res.status(200).json(createResp)
-}
-
-// Read page in collection
-async function readCollectionPage(req, res) {
-  const { accessToken } = req
-
-  const { siteName, pageName, collectionName, subcollectionName } = req.params
-  const { sha, content } = await CollectionController.ReadPage(
-    { siteName, accessToken },
-    { fileName: pageName, collectionName, thirdNavTitle: subcollectionName }
-  )
-
-  return res.status(200).json({ collectionName, pageName, sha, content })
-}
-
-// Update page in collection
-async function updateCollectionPage(req, res) {
-  const { accessToken } = req
-
-  const { siteName, pageName, collectionName, subcollectionName } = req.params
-  const { error } = UpdatePageRequestSchema.validate(req.body)
-  if (error) throw error
-  const { frontMatter, pageBody, sha, newFileName } = req.body
-  const updateResp = await CollectionController.UpdatePage(
-    { siteName, accessToken },
-    {
-      fileName: pageName,
+} = require("@validators/RequestSchema")
+
+class CollectionPagesRouter {
+  constructor({ collectionController }) {
+    this.collectionController = collectionController
+
+    const methods = Object.getOwnPropertyNames(Object.getPrototypeOf(this))
+    // Bind all methods
+    methods
+      .filter((method) => method !== "constructor")
+      .forEach((method) => {
+        this[method] = this[method].bind(this)
+      })
+  }
```

Comment:
I'm not entirely sure of the reason actually - the `this` object is undefined if we don't explicitly bind the class methods for the router. I suspect this is due to the express router intermediary - since the router methods are not called from the class (e.g. `CollectionPagesRouter.createCollectionPage`), the context of `this` isn't the class in the case of the router components only if we don't explicitly bind it - whereas all other classes have their methods directly invoked.

The auto-bind suggestion looks great, I'll make the change for it!

Reply:
Addressed in https://github.com/isomerpages/isomercms-backend/pull/234/commits/1c32873c4970dd9b35b60e10d03cc38b3c7a49d2